### PR TITLE
Fire extinguisher Bot

### PR DESCRIPTION
**Pull request by Ch0ubey**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,86 @@
-# Fire-Extinguisher-Bot
-Open Projects 2021
+# **Fire-Extinguisher-Bot**
```
Comment (vanshgoyal) [inline README.md]: Give space after open projects.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,86 @@
...
+![Bot Design](Images_videos/Images/Bot_view_1.jpg)
+<div align="center"><em>Bot View 1</em></div>
+<br/>
+
+![Bot Design 2](Images_videos/Images/Bot_view_4.PNG)
+<div align="center"><em>Bot View 2</em></div>
+<br/>
+
+## **Abstract**
+- Automatic Fire Extinguisher Robot is a Hardware-based model used to put out the fire during fire accidents automatically. The Bot has features to move in the direction of fire  and extinguish it automatically.
```
Comment (vanshgoyal) [inline README.md]: Justify all the text using paragraph tags. For eg.
`<p align="justify">`
  `<h2>Abstract</h2>`
  `<p>A Rube Goldberg Machine is a contraption intentionally designed to perform a simple task in an indirect and overly complicated way. The aim of this project is to depict the life of a college student through a similar complex assembly of mechanical and electronic components linked with each other creating a metaphor between humans and machines.</p>`
`</p>`

File: README.md [hunk 3]
```diff
@@ -1,2 +1,86 @@
...
+- The Bot finds its applications in rescue operations during fire accidents where concerned authorities can take time to enter the fire-prone areas.
+- Development of this autonomus bot using Controllers and image processing can help us to control fire without human inputs.Thus, minimising delays and unavailiblity of services.
+## Motivation
+The motivation behind this project essentially originated from our desire to learn new things. We can use this autonomous bot in case of laboratories or domestic fires, which can save lives or act as a quick remedy before the concerned authorities arrive. Although this technology is relatively unused in India, it can significantly impact the lives we lose to Fire incidents and our complete dependency on Emergency services.
+
+## Mechanical Aspect of Design
```
Comment (vanshgoyal) [inline README.md]: Elaborate your mech. aspect.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,86 @@
...
+
+## Limitations
+- Takes time to detect fire
+- Needs manual assistance while changing the fire extinguishers
+- Usable for household fires only
+## Future Improvement
+* Install cameras in every room
+* Link with smoke sensors in the area
+* Use custom made cylinders to avoid manual assistance
+## Team members
```
Comment (vanshgoyal) [inline README.md]: Hyperlink Github ids of team members and mentors.

**Review by vanshgoyal — COMMENTED** [review]
Please make following changes
1. Change the folder name of Images and Videos, Poster and report(according to the documentation)
2. Make the necessary changes suggested in the readme file.
3. In the src folder remove the bot programming pdf and insert the PID controller code separately.
4. Add IITR logo to poster and change the format to jpeg.
5. Follow uniform pattern for naming the folders and files. Eg. In the mech. design folder few are named using underscores where as few of them are separated by spaces.
6. create a new folder name simulation and insert the coppeliasim related files in the same.
7. Add applications header to readme.

File: src/nozzle_motor.ino [hunk 5]
```diff
@@ -0,0 +1,62 @@
+#include <Servo.h>
+#include <Math.h>
+
+int servoPin_y = 6;  //for rotation in vertical plane
+int servoPin_x = 5;  //for rotation in horizontal plane
+int pumpPin = 4;
+Servo Servo_y;
+Servo Servo_x; 
+void setup() {
```
Comment (vanshgoyal) [inline src/nozzle_motor.ino]: Add a new line and take care of indentation in the entire code base.

File: README.md (hunk 1 above)
Comment (Ch0ubey) [reply README.md]: do i hv to give a new line after "open projects 2021"??

**Comment by theobscuredev** [comment]
@Ch0ubey Lemme know once you've made the final changes. Make sure to tag me (@) in this thread.

**Comment by Ch0ubey** [comment]
OK sir

**Comment by theobscuredev** [comment]
@Ch0ubey & team,
Make the following changes ASAP and then we'll be good to go with the merge:

1. Justify the entire text in README using the `<p>` paragraph tag and `justify` attribute. Make sure you justify all the paragraphs separately. For eg:
`<p align="justify">`
  `<h2>Abstract</h2>`
  `<p>A Rube Goldberg Machine is .........................</p>`
`</p>`

Refer this README for the same: https://raw.githubusercontent.com/marsiitr/Rube-Goldberg-Machine/master/README.md

2. The 'Applications' section in your README is supposed to depict the applications of your robot and not the software used!! Try to mention the applications in the context of the parts and mechanisms used. For eg: You can explain how the actual prototype of the bot will benefit from the use of a particular mechanism/feature in real-life scenario. Refer previous year documentation (For eg: https://github.com/marsiitr/Underwater-Rover#applications) for more info.

3. Add a 'Total Cost' section in the Cost Structure table.

**Comment by theobscuredev** [comment]
@Ch0ubey Also, make sure you or someone from your team keeps updating this thread once you make any changes. Also, tag me once final changes have been made!

**Comment by Ch0ubey** [comment]
@theobscuredev sir now you can review it and tell us about the remaining issues (if any) . We tried to resolve all the queries raised by you and @vanshgoyal sir.

**Comment by Ch0ubey** [comment]
@theobscuredev sir from our side it's all done

**Comment by theobscuredev** [comment]
@Ch0ubey & team,
Still a few changes are required:

1. Few statements are still not justified in the README. Make sure you justify the entire text by splitting into appropriate paras.
2. Remove the Italic styling from 'Open Projects 2021' in your README. Simply use: ## Open Projects 2021
2. Information about belt, wheels, body materials and other components is missing in the cost structure. Update the same. Also, if there are any 3d printed parts, calculate their volume & try to guesstimate their price.
3. Include installation & execution instructions for your simulation files. Similar to this: https://github.com/marsiitr/RoverSim/tree/main/src/Dijkstra_Path_Planning#running-the-simulation
4. Change the heading from 'Mentees' to 'Team Members' in your poster.

Lemme know when you make these changes.

**Comment by Ch0ubey** [comment]
Ok sir

File: README.md [hunk 6]
```diff
@@ -1,2 +1,130 @@
...
+1. [Akanksha vijayvergiya](https://github.com/Akanksha-247)
+2. [Eeshan punde](https://github.com/eeshanpunde14)
+3. [Ganesh kalanidhi](https://github.com/GaneshK-RKE)
+4. [Mritunjay kumar choubey](https://github.com/Ch0ubey)
+5. [Poreddy sahith reddy](https://github.com/sahithreddyporeddy)
+6. [Srushti parbat](https://github.com/SrushtiParbat)
+## Mentors
+1. [Naman agarwalla](https://github.com/naman99-agar)
+2. [Parul chaudhary](https://github.com/Parul253)
+3. [Shubham goyal](https://github.com/shubham491981)
```
Comment (theobscuredev) [inline README.md]: Also capitalize the surname of each member/mentor. Make sure it's similar to the one mentioned in their Github profile.

Comment (Ch0ubey) [reply README.md]: ok sir :'(

**Comment by Ch0ubey** [comment]
@SanjeevKrishnan sir and @theobscuredev sir now you can review it.

**Review by SanjeevKrishnan — APPROVED** [review]
Great work team! Good effort
PR looks good. Merging the PR

**Pull request merged**